lib/absinthe/phase/document/result.ex

```diff
@@ -79,8 +79,13 @@ defmodule Absinthe.Phase.Document.Result do
     Map.merge(left, right, &do_deep_resolve/3)
   end
 
-  defp do_deep_resolve(_key, [%{} = left], [%{} = right]) do
-    [do_deep_merge(right, left)]
+  defp do_deep_resolve(_key, %{} = left, %{} = right) do
+    do_deep_merge(left, right)
+  end
+
+  defp do_deep_resolve(_key, [_|_] = left, [_|_] = right) when length(left) == length(right) do
```

Comment:
This seems like the wrong place to be doing that kind of checking. I'm gonna try to dig into this later and see if I can figure out if we can compute some of this stuff ahead of time so we don't have to do so much work at this point in the pipeline.